### PR TITLE
Code review exercise: Ben Walker

### PR DESCRIPTION
**Pull request by Bwalker01**

### CODE REVIEW

**Review by MichaelOH6 — APPROVED** [review]
Great Piece of work, couple of suggestions but I am happy with what you have coded

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java [hunk 1]
```diff
@@ -1,169 +1,152 @@
...
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+
+public class Game {
+	public static final int MAX_PLAYERS = 6;
+	public static final int STARTING_VALUE = 0;
+	public static final int AVAILABLE_SPACES = 11;
+	public static final int AMOUNT_OF_QUESTIONS = 50;
+
```
Comment (MichaelOH6) [inline exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:11]: This is something that I should've done. This is really useful and makes the code 10x more readable.
and also nice use of Arraylist

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Category.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.adaptionsoft.games.uglytrivia;
+
+import java.util.LinkedList;
+
+public class Category {
```
Comment (MichaelOH6) [inline exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Category.java:5]: Good use of another class to free up the Game class.

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java [hunk 3]
```diff
@@ -1,169 +1,152 @@
...
+	ArrayList players = new ArrayList();
+    int[] playerPlaces = new int[MAX_PLAYERS];
+    int[] playerCoins = new int[MAX_PLAYERS];
+    boolean[] inPenaltyBox  = new boolean[MAX_PLAYERS];
+
+	Category[] categories = {new Category("Pop"), new Category("Science"), new Category("Sports"), new Category("Rock")};
```
Comment (MichaelOH6) [inline exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:17]: good use of classes

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java [hunk 4]
```diff
@@ -1,169 +1,152 @@
...
+
+	}
+
+	private boolean isLeavingPenaltyBox(int roll) {
+		return roll % 2 != 0;
+	}
+
+
+	private void askQuestion() {
+		System.out.println(categories[playerPlaces[currentPlayer] % 4].askQuestion());
```
Comment (MichaelOH6) [inline exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:83]: maybe Nickpicking again but the playerPlaces is a tiny bit confusing in this example

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java [hunk 5]
```diff
@@ -1,169 +1,152 @@
...
+	}
+	
+	
+	private String currentCategory() {
+		switch(playerPlaces[currentPlayer]) {
```
Comment (MichaelOH6) [inline exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:88]: Nice use of switch statements, more readable than a load of if statements, however maybe try to work out a way to not use numbers.

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java [hunk 6]
```diff
@@ -1,169 +1,152 @@
...
+			case 1:
+			case 5:
+			case 9:
+				return "Science";
+			case 2:
+			case 6:
+			case 10:
+				return "Sports";
+			default:
+				return "Rock";
```
Comment (MichaelOH6) [inline exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:102]: Maybe throw a out of field error instead of having rock as a default

**Review by MichaelOH6 — APPROVED** [review]

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Category.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    public LinkedList<String> getQuestions() {
+        return questions;
+    }
+
+    public String askQuestion() {
+        return questions.removeFirst();
+    }
+
+    public void addQuestion(int num) {
+        questions.addLast(String.format("%s Question %d", name, num));
```
[the remark this answers is not in the document]
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Category.java:26]: Any suggestions on variable names for each?

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java [hunk 8]
```diff
@@ -1,169 +1,152 @@
...
+    int[] playerCoins = new int[MAX_PLAYERS];
+    boolean[] inPenaltyBox  = new boolean[MAX_PLAYERS];
+
+	Category[] categories = {new Category("Pop"), new Category("Science"), new Category("Sports"), new Category("Rock")};
+
+    int currentPlayer = 0;
+    boolean leavingPenaltyBox;
+    
+    public  Game(){
+    	for (int count = 0; count < AMOUNT_OF_QUESTIONS; count++) {
```
[the remark this answers is not in the document]
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:23]: Thank you

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java (hunk 1 above)
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:11]: Thank you

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Category.java (hunk 2 above)
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Category.java:5]: Thank you

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java (hunk 3 above)
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:17]: Thank you

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java (hunk 6 above)
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:102]: Would you suggest that a `IllegalArgumentException` would be the best approach for an error

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java (hunk 4 above)
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:83]: Would you use a temp variable around `playerPlaces[currentPlayer]` or would you use a method call to work out the value of `playerPlaces[currentPlayer] % 4`

File: exercises/java/src/main/java/com/adaptionsoft/games/trivia/runner/GameRunner.java [hunk 9]
```diff
@@ -10,22 +10,22 @@ public class GameRunner {
 	private static boolean notAWinner;
 
 	public static void main(String[] args) {
-		Game aGame = new Game();
-		
-		aGame.add("Chet");
-		aGame.add("Pat");
-		aGame.add("Sue");
+		Game mainGame = new Game();
```
[the remark this answers is not in the document]
Comment (Bwalker01) [reply exercises/java/src/main/java/com/adaptionsoft/games/trivia/runner/GameRunner.java:13]: Thank you

File: exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java (hunk 5 above)
Comment (mrmanc) [reply exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java:88]: You might be interested to hear that there is a way to still very clearly represent the way the board is laid out, and have an even lower complexity score (of 1). Bit of a challenge 😃

**Pull request closed without merging**